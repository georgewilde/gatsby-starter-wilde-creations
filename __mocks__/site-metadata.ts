import { SiteMetadata } from '@context/site-metadata';

export const SiteMetadataMock: SiteMetadata = {
  companyName: 'Acme Co.',
  description: '',
  author: '',
};
