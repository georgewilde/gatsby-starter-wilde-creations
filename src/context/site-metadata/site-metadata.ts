import { createContext } from 'react';

export interface SiteMetadata {
  companyName: string;
  description: string;
  author: string;
}

const defaultState: SiteMetadata = {
  companyName: '',
  description: '',
  author: '',
};

export const SiteMetadataContext = createContext(defaultState);
