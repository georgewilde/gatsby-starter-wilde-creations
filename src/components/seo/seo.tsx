import React, { FC } from 'react';
import { Helmet } from 'react-helmet';
import { SiteMetadataContext } from '@context/site-metadata';

interface SEOProps {
  title: string;
  description?: string;
  lang?: string;
  meta?: any[];
}

export const SEO: FC<SEOProps> = ({
  title,
  description = '',
  lang = 'en',
  meta = [],
}) => {
  return (
    <SiteMetadataContext.Consumer>
      {(siteMetadata): JSX.Element => {
        const metaDescription = description || siteMetadata.description;

        return (
          <Helmet
            htmlAttributes={{
              lang,
            }}
            title={title}
            titleTemplate={`%s | ${siteMetadata.companyName}`}
            meta={[
              {
                name: `description`,
                content: metaDescription,
              },
              {
                property: `og:title`,
                content: title,
              },
              {
                property: `og:description`,
                content: metaDescription,
              },
              {
                property: `og:type`,
                content: `website`,
              },
              {
                name: `twitter:card`,
                content: `summary`,
              },
              {
                name: `twitter:creator`,
                content: siteMetadata.author,
              },
              {
                name: `twitter:title`,
                content: title,
              },
              {
                name: `twitter:description`,
                content: metaDescription,
              },
            ].concat(meta)}
          />
        );
      }}
    </SiteMetadataContext.Consumer>
  );
};
