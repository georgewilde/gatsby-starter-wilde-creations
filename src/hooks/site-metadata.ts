import { graphql, useStaticQuery } from 'gatsby';

import { SiteMetadata } from '@context/site-metadata';

interface SiteMetadataResult {
  site: {
    siteMetadata: SiteMetadata;
  };
}

export const useSiteMetadata = (): SiteMetadata => {
  const { site } = useStaticQuery<SiteMetadataResult>(
    graphql`
      query SiteMetaData {
        site {
          siteMetadata {
            companyName
            description
            author
          }
        }
      }
    `
  );

  return site.siteMetadata;
};
