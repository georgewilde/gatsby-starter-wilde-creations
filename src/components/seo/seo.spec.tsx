import React from 'react';
import renderer from 'react-test-renderer';

import { SiteMetadata, SiteMetadataContext } from '@context/site-metadata';
import { SEO } from '@components/seo/seo';

describe(`SEO component`, () => {
  let siteMetadata: SiteMetadata;

  beforeEach(() => {
    siteMetadata = {
      companyName: 'Acme Co.',
      description: '',
      author: '',
    };
  });

  it(`should not render anything to the screen`, () => {
    const tree = renderer
      .create(
        <SiteMetadataContext.Provider value={siteMetadata}>
          <SEO title="test title" />
        </SiteMetadataContext.Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
