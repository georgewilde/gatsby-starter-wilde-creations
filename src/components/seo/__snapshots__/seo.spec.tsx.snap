// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEO component should not render anything to the screen 1`] = `null`;
